front_end/panels/timeline/TimelinePanel.ts

```diff
@@ -731,6 +744,31 @@ export class TimelinePanel extends UI.Panel.Panel implements Client, TimelineMod
     });
   }
 
+  private async rnPrepareForTraceCapturedInBackground(): Promise<void> {
+    this.setUIControlsEnabled(false);
+
+    if (this.statusPane) {
+      this.statusPane.finish();
+      this.statusPane.updateStatus(i18nString(UIStrings.stoppingTimeline));
+      this.statusPane.updateProgressBar(i18nString(UIStrings.received), 0);
+    }
+    this.setState(State.STOP_PENDING);
+
+    const rootTarget = SDK.TargetManager.TargetManager.instance().rootTarget();
+    const primaryPageTarget = SDK.TargetManager.TargetManager.instance().primaryPageTarget();
+    if (!primaryPageTarget) {
```

Comment:
Flip order of checks and put them directly after grabbing their reference maybe?